File: array/171.py
from typing import List


def distributeCandies(candyType: List[int]) -> int:
    n = len(candyType)
    ans = [0] * 200001

    for candy in candyType:
        if candy >= 0:
            ans[candy] += 1
        else:
            ans[abs(candy) + 100000] += 1

    cont = 0
    for elem in ans:
        if elem > 0:
            cont += 1

    return min(cont, n // 2)


print(distributeCandies([1, 1, 2, 2, 3, 3, -100000, -9999, 999, 9000, 9000, 100000]))

File: array/171.cpp
#include <bits/stdc++.h>

using namespace std;

int distributeCandies(vector<int>& candyType) {
    vector<int> ans(200001, 0);
    int n = candyType.size();

    for(int i = 0; i < candyType.size(); ++i) {
        if (candyType[i] >= 0)
            ans[candyType[i]] += 1;
        else
            ans[abs(candyType[i]) + 10000] += 1;
    }

    int cont = 0;
    for(int i =0;i < ans.size(); ++i)
        cont += (ans[i] >= 1);


    return min(n / 2, cont);
}

int main() {
    vector<int> candyType = {100000,0,100000,0,100000,0,100000,0,100000,0,100000,0};
    cout<<distributeCandies(candyType);

    return 0;
}